src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetAnnotationExecutor.java

```diff
@@ -51,19 +51,31 @@ public void execLoad(BulkLoadFile bulkLoadFile) {
 
 			bulkLoadFileDAO.merge(bulkLoadFile);
 			List<Long> datasetIdsLoaded = new ArrayList<>();
+			List<Long> htpAnnotaionsIdsLoaded = new ArrayList<>();
 			BulkLoadFileHistory history = new BulkLoadFileHistory(htpExpressionDatasetData.getData().size());
-			createHistory(history, bulkLoadFile);
-			boolean result = runLoaddatasetid(externalDataBaseEntityService, history, dataProvider, htpExpressionDatasetData.getData(), datasetIdsLoaded, false);
+			
+			runLoad(history, dataProvider, htpExpressionDatasetData.getData(), htpAnnotaionsIdsLoaded, datasetIdsLoaded);
+			
+			history.finishLoad();
+			updateHistory(history);
 		} catch (Exception e) {
 			failLoad(bulkLoadFile, e);
 			e.printStackTrace();
 		}
 	}
 
-	private boolean runLoaddatasetid(ExternalDataBaseEntityService externalDataBaseEntityService, BulkLoadFileHistory history, BackendBulkDataProvider dataProvider, List<HTPExpressionDatasetAnnotationFmsDTO> htpDatasetData, List<Long> datasetIdsLoaded, boolean isUpdate) {
+	private void runLoad(BulkLoadFileHistory history, BackendBulkDataProvider dataProvider, List<HTPExpressionDatasetAnnotationFmsDTO> htpDataset, List<Long> htpAnnotaionsIdsLoaded, List<Long> datasetIdsLoaded) {
 		ProcessDisplayHelper ph = new ProcessDisplayHelper();
 		ph.addDisplayHandler(loadProcessDisplayService);
-		ph.startProcess("External Database Entity DTO Update for " + dataProvider.name(), htpDatasetData.size());
+		ph.startProcess("HTP Expression Dataset Annotation DTO Update for " + dataProvider.name(), htpDataset.size() * 2);
+
+		runLoaddatasetid(externalDataBaseEntityService, history, dataProvider, htpDataset, datasetIdsLoaded, ph);
+		runLoadHtpAnnotations(htpExpressionDatasetAnnotationService, history, dataProvider, htpDataset, htpAnnotaionsIdsLoaded, ph);
+
+		ph.finishProcess();
+	}
+
+	private void runLoaddatasetid(ExternalDataBaseEntityService externalDataBaseEntityService, BulkLoadFileHistory history, BackendBulkDataProvider dataProvider, List<HTPExpressionDatasetAnnotationFmsDTO> htpDatasetData, List<Long> datasetIdsLoaded,  ProcessDisplayHelper ph) {
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.SingleSpaceSeparatorCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Use a single space to separate non-whitespace characters.
<!-- __reviewdog__:ChBjZTc1Y2JmYjY5Mjc2NTFjEgpjaGVja3N0eWxl -->